src/components/toast/toast.pw.tsx

```diff
@@ -0,0 +1,355 @@
+
+    await expect(icon).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(icon).toHaveCSS("outline", "rgba(0, 0, 0, 0) solid 3px");
+  });
+
+  test("should render Toast component with focused close icon and correct style when focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(<ToastComponent variant="notice" />, {
+      hooksConfig: { focusRedesignOptOut: true },
+    });
+
+    const icon = closeIconButton(page);
+    await icon.focus();
+
+    await expect(icon).toHaveCSS("outline", "rgb(255, 188, 25) solid 3px");
+  });
+});
+
+(["left", "center", "right"] as const).forEach((align) => {
+  test(`should render Toast component with align prop set to ${align}`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ToastComponent align={align} isCenter={false} />);
+
+    await expect(toastComponent(page).locator("..").locator("..")).toHaveCSS(
+      "justify-content",
+      align
+    );
+  });
+});
+
+(["top", "bottom"] as const).forEach((alignY) => {
+  test(`should render Toast component with alignY prop set to ${alignY}`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ToastComponent alignY={alignY} />);
+
+    await expect(toastComponent(page)).toHaveCSS(alignY, "0px");
+  });
+});
+
+test("clicking close icon should close toast when a new modal has opened on top of it", async ({
+  mount,
+  page,
+}) => {
+  await mount(<ToastWhenOtherModalRenders />);
+
+  await button(page).nth(0).click();
+  await expect(toastComponent(page)).toBeVisible();
+
+  await button(page).nth(1).click();
+  await expect(alertDialogPreview(page)).toBeVisible();
+
+  const closeIconToast = closeIconButton(page).nth(0);
+  await closeIconToast.click();
+
+  await expect(toastComponent(page)).not.toBeVisible();
+});
+
+test.describe("check events for Toast component", () => {
+  test("should call onDismiss callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <ToastComponent
+        onDismiss={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    await closeIconButton(page).click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onDismiss callback when a ESC event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <ToastComponent
+        onDismiss={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    await page.keyboard.press("Escape");
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("check accessibility for Toast component", () => {
+  ([
+    TOAST_COLORS[0],
+    TOAST_COLORS[1],
+    TOAST_COLORS[2],
+    TOAST_COLORS[3],
+  ] as ToastProps["variant"][]).forEach((variant) => {
+    test(`should render Toast component with ${variant} variant and check accessibility`, async ({
```

Comment:
```suggestion
    test(`should render with ${variant} variant and check accessibility`, async ({
```

**suggestion** (non-blocking): we shouldn't repeat "should render Toast component" in every test, especially when the `describe` message mentions the component name.

Same applies to all of these tests.